pkg/workspace/data_values_pre_processing.go

```diff
@@ -37,46 +38,89 @@ func (o DataValuesPreProcessing) Apply() (*DataValues, []*DataValues, error) {
 }
 
 func (o DataValuesPreProcessing) apply(files []*FileInLibrary) (*DataValues, []*DataValues, error) {
-	values := o.loader.schema.AsDataValue()
-	var libraryValues []*DataValues
-	for _, fileInLib := range files {
-		valuesDocs, err := o.templateFile(fileInLib)
-		if err != nil {
-			return nil, nil, fmt.Errorf("Templating file '%s': %s", fileInLib.File.RelativePath(), err)
-		}
+	_, checkSchema := o.loader.schema.(*schema.DocumentSchema)
 
-		for _, valuesDoc := range valuesDocs {
-			dv, err := NewDataValues(valuesDoc)
+	allDvs, err := o.collectDataValuesDocs(files)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// merge all Data Values YAML documents into one
+	var dvsForOtherLibraries []*DataValues
+	var dataValuesDoc *yamlmeta.Document
```

Comment:
i wonder if naming this from dataValuesDoc -> resultDVsDoc would make it a bit more readable. (it's the result from this loop)

Reply:
... and taking that suggestion a step further, if we use the same naming scheme across identifiers: `(qualifier)(noun)`, the eye parses identifiers using the same pattern matcher, reducing cognitive load.

Example:
- `allDvs`
- `otherLibraryDvs`
- `resultDvDoc`
- others?